core/dbt/cli/main.py

```diff
@@ -341,9 +341,13 @@ def list(ctx, **kwargs):
 @p.version_check
 @p.write_manifest
 @requires.preflight
+@requires.profile
+@requires.project
+@requires.runtime_config
+@requires.manifest(write_perf_info=True)
 def parse(ctx, **kwargs):
     """Parses the project and provides information on performance"""
-    click.echo(f"`{inspect.stack()[0][3]}` called\n flags: {ctx.obj['flags']}")
+    # manifest generation and writing happens in @requires.manifeset
     return None, True
```

Comment:
Thank you for the context. I had spoken with Gerda about the function of the existing `ParseTask` class, and she had said I could take out the tracking. Adding it back in shouldn't be difficult, and could be done conditionally.

For your proposal of returning the manifest from the task directly, I think that's something we need to discuss on the exec team. I believe we should have a standard output for each of these functions, perhaps a dict with a `result` or `message` or whatever key should contain result, in case we want to add meta information to that result object later.

Reply:
> I had spoken with Gerda about the function of the existing `ParseTask` class, and she had said I could take out the tracking. Adding it back in shouldn't be difficult, and could be done conditionally.

This is fine with me too. Gerda consolidated all the events in this task, anyway, to just `ParseCmdOut` (in the `main` branch). We'll just want to remove that event now that it's no longer being called anywhere. For simplicity, let's track that in a separate issue, rather than try to do branch merging shenanigans here.

---

The point I was making above was around, `dbt parse --compile` would also conditionally include the step around manifest compilation / graph generation, which can be very slow (as we've seen in recent reports/issues from folks with large projects). Let's track that in the new issue, "Abstract manifest compilation / graph generation from tasks."

---

> For your proposal of returning the manifest from the task directly, I think that's something we need to discuss on the exec team. I believe we should have a standard output for each of these functions, perhaps a dict with a `result` or `message` or whatever key should contain result, in case we want to add meta information to that result object later.

Agreed! Let's keep discussing in the separate linked issue. The biggest considerations are:
- How to keep the `results` _relatively_ consistent for all commands (although different tasks already return differently typed result objects)
- Should we expose the full manifest, as part of our public API? Just a part of it? Should we call this "experimental" functionality (liable to change in future versions)? We have the power to decide these things!

Reply:
We might want to call these changes out here:
https://docs.getdbt.com/guides/migration/versions/upgrading-to-v1.5

Did the proposed **"Abstract manifest compilation / graph generation from tasks."** issue ever get created?

In the meantime, I'm going to delete both of these from `params.py` as part of https://github.com/dbt-labs/dbt-core/issues/6546 since they don't appear to do anything:

https://github.com/dbt-labs/dbt-core/blob/3f76f82c88ad35436d9c7e35d27402d3fe3b892c/core/dbt/cli/params.py#L44-L49

https://github.com/dbt-labs/dbt-core/blob/3f76f82c88ad35436d9c7e35d27402d3fe3b892c/core/dbt/cli/params.py#L491-L496